common/types.go

```diff
@@ -22,35 +22,53 @@ package common
 
 import (
 	"bytes"
+	"encoding/binary"
 	"encoding/hex"
 	"errors"
 	"fmt"
 	"math/big"
 	"math/rand"
 	"reflect"
+	"sync/atomic"
+	"time"
 
 	"github.com/klaytn/klaytn/common/hexutil"
 	"github.com/klaytn/klaytn/crypto/sha3"
 )
 
 const (
-	HashLength      = 32
-	AddressLength   = 20
-	SignatureLength = 65
+	HashLength         = 32
+	ExtHashNonceLength = 6
+	ExtHashLength      = HashLength + ExtHashNonceLength
+	AddressLength      = 20
+	SignatureLength    = 65
 )
 
 var (
 	hashT    = reflect.TypeOf(Hash{})
+	extHashT = reflect.TypeOf(ExtHash{})
 	addressT = reflect.TypeOf(Address{})
 )
 
-var lastPrecompiledContractAddressHex = hexutil.MustDecode("0x00000000000000000000000000000000000003FF")
-
 var (
+	lastPrecompiledContractAddressHex = hexutil.MustDecode("0x00000000000000000000000000000000000003FF")
+
+	extHashCounterMin  = uint64(0x100) // Any counter below extHashCounterMin are reserved.
+	extHashCounter     = extHashCounterMin
+	extHashRootNonce   = ExtHashNonce{0, 0, 0, 0, 0, 0x11}
+	extHashLegacyNonce = ExtHashNonce{0, 0, 0, 0, 0, 0x45}
```

Comment:
@ethan-kr We need rationales for these numbers in KIP-112

Reply:
I just needed a magic number to differentiate, and I put my personal favorite number.